src/main/java/com/gotocompany/firehose/sink/common/blobstorage/oss/ObjectStorageService.java

```diff
@@ -0,0 +1,55 @@
+package com.gotocompany.firehose.sink.common.blobstorage.oss;
+
+import com.aliyun.oss.OSS;
+import com.aliyun.oss.OSSClientBuilder;
+import com.aliyun.oss.OSSException;
+import com.aliyun.oss.model.ObjectMetadata;
+import com.gotocompany.firehose.config.OSSConfig;
+import com.gotocompany.firehose.sink.common.blobstorage.BlobStorage;
+import com.gotocompany.firehose.sink.common.blobstorage.BlobStorageException;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.nio.file.Paths;
+
+public class ObjectStorageService implements BlobStorage {
+    private final OSS ossClient;
+    private final String bucketName;
+    private final String directoryPrefix;
+
+    public ObjectStorageService(OSSConfig config) {
+        this.ossClient = createOSSClient(
+                config.getOSSEndpoint(),
+                config.getOSSAccessKeyId(),
+                config.getOSSAccessKeySecret()
+        );
+        this.bucketName = config.getOSSBucketName();
+        this.directoryPrefix = config.getOSSDirectoryPrefix();
+    }
+
+    protected OSS createOSSClient(String endpoint, String accessKeyId, String accessKeySecret) {
+        return new OSSClientBuilder().build(endpoint, accessKeyId, accessKeySecret);
```

Comment:
Should we introduce the option to pass configurable socketTimeout, connectionTimeout, connectionRequestTimeout and requestTimeout? There is also an option to pass configurable retry strategy for the OSSClient

Reply:
since we are exposing connection parameters in config, the builder should accommodate the values